### PR TITLE
Feature/1624/add discussion link for tools

### PR DESCRIPTION
**Pull request by nrzhao4**
- for [DOCK-1624](https://ucsc-cgl.atlassian.net/browse/DOCK-1624) 
- added forumurl column for tools
- `/containers/{containerId}` updates the forumurl

### CODE REVIEW

File: dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java [hunk 1]
```diff
@@ -1313,6 +1313,29 @@ public void testUpdateToolPathDockerfile() throws ApiException {
...
+        toolsApi.refresh(toolTest.getId());
+
+        //change the forumurl
+        toolTest.setForumUrl("hello.com");
+        toolsApi.updateContainer(toolTest.getId(), toolTest);
+        toolsApi.refresh(toolTest.getId());
+
+        //check the tool's forumurl is updated in the database
+        final String updatedForumUrl = testingPostgres.runSelectStatement("select forumurl from tool where id = " + toolTest.getId(), String.class);
+        assertEquals("the forumurl should be hello.com", "hello.com", updatedForumUrl);
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java:1336]: can use constant here and on line 1330 to make it really obvious what's going on

**Review by denis-yuen — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

File: dockstore-webservice/src/main/resources/migrations.1.10.0.xml [hunk 2]
```diff
@@ -119,4 +119,9 @@
         <modifyDataType columnName="forumurl" newDataType="varchar(256)" tableName="service"/>
         <modifyDataType columnName="forumurl" newDataType="varchar(256)" tableName="workflow"/>
     </changeSet>
+    <changeSet author="nzhao (generated)" id="toolForumUrl">
```
Comment (coverbeck) [inline dockstore-webservice/src/main/resources/migrations.1.10.0.xml:122]: I think this should be part of a new migrations.1.11.0.xml, unless we're targeting for a hotfix. Adding a new migrations file is a bit of a pain; you also have to at least update compose_setup, propose_migrations.sh, and the integration tests. Just pointing it out, will defer to @denis-yuen  and @garyluu

Comment (garyluu) [reply dockstore-webservice/src/main/resources/migrations.1.10.0.xml:122]: ya, I agree with 1.11.0

**Review by wshands — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

**Comment by garyluu** [comment]
Oops, approved by mistake

**Review by garyluu — COMMENTED** [review]
Waiting until Charle's comment is done

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java [hunk 3]
```diff
@@ -505,7 +505,7 @@ public void testUpdateUserMetadataFromGithub() {
         System.out.println(usersApi.getUser().getUserProfiles().get("github.com"));
         assertNull(userProfile.getName());
         assertEquals("dockstore.test.user2@gmail.com", userProfile.getEmail());
-        assertEquals("https://avatars1.githubusercontent.com/u/17859829?v=4", userProfile.getAvatarURL());
+        assertTrue(userProfile.getAvatarURL().endsWith("githubusercontent.com/u/17859829?v=4"));
```
Comment (coverbeck) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java:508]: Saw your comments on Slack... Just curious what changed? It's a random digit after `avatars`, or a different one? Or the subdomain isn't there at all?

**Review by coverbeck — APPROVED** [review]

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java (hunk 3 above)
Comment (denis-yuen) [reply dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java:508]: `https://avatars1` lost the `1`

**Pull request merged**